dipy/io/stateful_tractogram.py

```diff
@@ -393,11 +393,11 @@ def data_per_streamline(self, data):
 
     def get_data_per_point_keys(self):
         """ Return a list of the data_per_point attribute names """
-        return list(self.data_per_point.keys())
+        return set(self.data_per_point.keys())
 
     def get_data_per_streamline_keys(self):
         """ Return a list of the data_per_streamline attribute names """
-        return list(self.data_per_streamline.keys())
+        return set(self.data_per_streamline.keys())
```

Comment:
same comment as above: can you use ` list(set(...))` to keep the same behavior as before